exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/BatchValidator.java

```diff
@@ -150,57 +424,63 @@ private int validateOffsetVector(String name, UInt4Vector offsetVector, int valu
+            "Offset vector [%d] contained %d, expected >= %d",
+            i, offset, prevOffset));
       } else if (offset > maxOffset) {
-        error(name, offsetVector, "Invalid offset at index " + i + " = " + offset + " exceeds maximum of " + maxOffset);
+        error(name, offsetVector, String.format(
+            "Invalid offset at index %d: %d exceeds maximum of %d",
+            i, offset, maxOffset));
       }
       prevOffset = offset;
     }
     return prevOffset;
   }
 
   private void error(String name, ValueVector vector, String msg) {
-    if (errorCount == 0) {
-      logger.error("Found one or more vector errors from " + batch.getClass().getSimpleName());
-    }
-    errorCount++;
-    if (errorCount >= MAX_ERRORS) {
-      return;
-    }
-    String fullMsg = "Column " + name + " of type " + vector.getClass().getSimpleName( ) + ": " + msg;
-    logger.error(fullMsg);
-    if (errorList != null) {
-      errorList.add(fullMsg);
-    }
+    errorReporter.error(name, vector, msg);
   }
 
-  private void validateNullableVector(String name, NullableVector vector) {
-    // Can't validate at this time because the bits vector is in each
-    // generated subtype.
-
-    // Validate a VarChar vector because it is common.
-
-    if (vector instanceof NullableVarCharVector) {
-      VarCharVector values = ((NullableVarCharVector) vector).getValuesVector();
-      validateVarCharVector(name + "-values", values, rowCount);
+  private void verifyIsSetVector(ValueVector parent, UInt1Vector bv) {
+    final String name = String.format("%s (%s)-bits",
+        parent.getField().getName(),
+        parent.getClass().getSimpleName());
+    final int rowCount = parent.getAccessor().getValueCount();
+    final int bitCount = bv.getAccessor().getValueCount();
+    if (bitCount != rowCount) {
+      error(name, bv, String.format(
+          "Value count = %d, but bit count = %d",
+          rowCount, bitCount));
+    }
+    final UInt1Vector.Accessor ba = bv.getAccessor();
+    for (int i = 0; i < bitCount; i++) {
+      final int value = ba.get(i);
+      if (value != 0 && value != 1) {
+        error(name, bv, String.format(
+            "%s %s: bit vector[%d] = %d, expected 0 or 1",
+            i, value));
+      }
     }
-  }
-
-  private void validateFixedWidthVector(String name, FixedWidthVector vector) {
-    // TODO Auto-generated method stub
-
   }
 
   /**
-   * Obtain the list of errors. For use in unit-testing this class.
-   * @return the list of errors found, or null if error capture was
-   * not enabled
+   * Print a record batch. Uses code only available in a test build.
+   * Classes are not visible to the compiler; must load dynamically.
+   * Does nothing if the class is not available.
    */
 
-  public List<String> errors() { return errorList; }
+  public static void print(RecordBatch batch) {
+    try {
+      final Class<?> helper = Class.forName("org.apache.drill.test.rowSet.RowSetUtilities");
```

Comment:
Checked `org.apache.drill.test.rowSet.RowSetUtilities` it does not have `print` method. Could you please check?